### PR TITLE
[MM-54800] Convert "close_channel.tsx" from Class Component to Function Component

### PR DESCRIPTION
**Pull request by ARJ2160**
Fixed - Issue-[#24764](https://github.com/mattermost/mattermost/issues/24764)

New Function Component
image

Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by ARJ2160** [comment]
I am not so sure about the type assertions in the test file
Do you think the assertions are necessary?
The file doesn't show any error without them so we can remove them without it being a problem.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.test.tsx [hunk 1]
```diff
@@ -16,30 +16,36 @@ describe('components/ChannelHeaderDropdown/MenuItem.CloseChannel', () => {
         },
     };
 
-    it('should match snapshot', () => {
-        const wrapper = shallow<CloseChannel>(<CloseChannel {...baseProps}/>);
+    it("should match snapshot", () => {
+        const wrapper = shallow<typeof CloseChannel>(
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.test.tsx:20]: You can remove the type assertion.

```suggestion
        const wrapper = shallow(
```

File: webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.test.tsx [hunk 2]
```diff
@@ -16,30 +16,36 @@ describe('components/ChannelHeaderDropdown/MenuItem.CloseChannel', () => {
         },
     };
 
-    it('should match snapshot', () => {
-        const wrapper = shallow<CloseChannel>(<CloseChannel {...baseProps}/>);
+    it("should match snapshot", () => {
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.test.tsx:19]: We use single quotes instead of double quotes, so probably this will fail on the linter.

You can fix all linter issues by running `npm run fix` in the channels folder.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx [hunk 3]
```diff
@@ -7,25 +7,31 @@ import Menu from 'components/widgets/menu/menu';
...
-                text={localizeMessage('center_panel.archived.closeChannel', 'Close Channel')}
-            />
-        );
-    }
-}
+    return (
+        <Menu.ItemAction
+            show={isArchived}
+            onClick={handleClose}
+            text={localizeMessage(
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx:29]: `localizeMessage` is a function we want to get rid of. In this case you should use `intl.formatMessage`, and for that, you should do `const intl = useIntl();` in the component.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx [hunk 4]
```diff
@@ -7,25 +7,31 @@ import Menu from 'components/widgets/menu/menu';
...
 }
 
-export default class CloseChannel extends React.PureComponent<Props> {
-    private handleClose = () => {
-        this.props.actions.goToLastViewedChannel();
+const CloseChannel = ({
+    isArchived,
+    actions,
+}: CloseChannelProps): JSX.Element => {
+    const handleClose = () => {
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx:21]: This is creating a new function on every render, potentially re-rendering the underlying component. We could use `useCallback` here, but the callback is so simple, we can use it directly as a prop for the next component. i.e.

```
<Menu.ItemAction
    onClick={actions.goToLastViewedChannel}
/>
```

File: webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx [hunk 5]
```diff
@@ -7,25 +7,31 @@ import Menu from 'components/widgets/menu/menu';
 
 import {localizeMessage} from 'utils/utils';
 
-type Props = {
+interface CloseChannelProps {
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx:10]: The name change is not needed. There is only one component in this file, and the props are only used in this file, so it is inferred that the props are of the CloseChannel component.

Comment (larkox) [inline webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.tsx:10]: Why the change from type to interface?

**Review by larkox — CHANGES_REQUESTED** [review]
Thank you for the contribution! Since the Issue was assigned to another person, I am not sure if we will merge this PR (I expect the other person to be OK to move on with this PR, but still I have to ask), but here is some feedback.

File: webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.test.tsx (hunk 1 above)
Comment (ARJ2160) [reply webapp/channels/src/components/channel_header_dropdown/menu_items/close_channel/close_channel.test.tsx]: alright

**Comment by harshilsharma63** [comment]
@ARJ2160 can you look into failing tests?

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]
CI tests need to be fixed

**Comment by ARJ2160** [comment]
Actually, I'm new to Jest. Could you help me with this? My machine can't run these tests without crashing, so I can't verify any solution I came up with without pushing a commit.

**Comment by larkox** [comment]
@ARJ2160 In theory, all you have to do is to run `npm run test -- -u -- path/to/test/file.ts` in the channels folder. That will update the snapshots.

**Comment by ARJ2160** [comment]
Thank you so much @larkox !
All tests under channel_header_dropdown are passing now

**Review by harshilsharma63 — APPROVED** [review]
👌👌👌

**Review by larkox — APPROVED** [review]
Great work! Thanks!

**Review by lindy65 — APPROVED** [review]
Thanks @ARJ2160 - looks good to me 👍 

cc @larkox
